src/material/button/_button-base.scss

```diff
@@ -43,10 +43,15 @@
 
   // The content should appear over the state and ripple layers, otherwise they may adversely affect
   // the accessibility of the text content.
-  .mdc-button__label {
+  .mdc-button__label,
+  .mat-icon {
     z-index: 1;
   }
 
+  .mat-icon {
+    position: relative;
```

Comment:
Can you also move this into the label selector? Technically it's already `position: relative` so it'll be a noop, but we'll save one CSS rule.